doc/develop/modules.rst

```diff
@@ -86,19 +86,50 @@ Module Repositories
-  repository should preserve the upstream repository folder structure.
+  * The module repository should preserve the upstream repository folder structure.
+
+  * Follow one of these strategies to track the upstream repository:
+
+    * The preferred strategy is that the module tracks directly the upstream main development
+      branch or maintenance branches which are pointed by the Zephyr manifest without any changes,
+      where all Zephyr specific code is hosted by the upstream project.
+      Pull requests are sent directly to the upstream repository.
+      Only in exceptional circumstances, when a bug fix is necessary faster than the upstream is
+      able to merge it, should the downstream, Zephyr module repository branch, be forked in a
+      separate Zephyr specific branch. This should only be done for the shortest period of time
+      possible and with the intention of returning to the upstream tracked branch as soon as
+      possible.
+
+    * For modules tracking too unresponsive upstream projects, we may need to have longer
+      lived Zephyr branches. Still, all changes in Zephyr specific branches should be submitted
+      for review to the upstream project. With the aim of abandoning these Zephyr downstream
+      branches as soon as possible.
+
+    * In some cases the upstream may not be willing to accept some Zephyr specific files or changes.
+      This is an undesirable situation as it forces us to maintain a permanent fork of the upstream
+      repository. In this case, two options are possible:
+
+      1. Relatively short lived, Zephyr specific branches are kept downstream with the Zephyr
+         specific patches. Every now and then, the module is resynchronized with the upstream
+         by creating a new branch downstream forking from the updated upstream tracked branch and
+         rebasing the old downstream forked branch.
+
+      2. A permanently forked 'zephyr' branch, in which, periodically, patches which made it both to
+         this branch and the tracked upstream are reverted and the upstream main is merged.
+
+      Still in either of these cases, we should always aim at minimizing the fork, patches
+      should be submitted upstream, and after they are merged, possible downstream equivalent
+      changes be replaced in favor of the upstream accepted ones.
 
   .. note::
 
-     It is not required in module repositories to maintain a 'master'
-     branch mirroring the master branch of the external repository. It
-     is not recommended as this may generate confusion around the module's
-     main branch, which should be 'zephyr'.
+     Under no circumstances may the Zephyr downstream fork have forked branches with the same name
+     as the upstream tracked branches. That is, for example, the downstream may not have a 'main'
+     with different commits than the upstream 'main'.
+
+* Modules should name Zephyr specific branches starting with the 'zephyr\_' prefix.
```

Comment:
@nordicjm We will have to rename the synchronization branch, by adding the prefix.